code/game/objects/items/storage/fancy.dm

```diff
@@ -218,13 +218,40 @@
 
 /obj/item/storage/fancy/cigarettes/Initialize(mapload)
 	. = ..()
-	atom_storage.quickdraw = TRUE
+	atom_storage.display_contents = FALSE
 	atom_storage.set_holdable(list(/obj/item/clothing/mask/cigarette, /obj/item/lighter))
+	register_context()
+
+/obj/item/storage/fancy/cigarettes/attack_hand_secondary(mob/user, list/modifiers)
+	. = ..()
+	quick_remove_item(/obj/item/clothing/mask/cigarette, user)
+
+/obj/item/storage/fancy/cigarettes/AltClick(mob/user)
+	. = ..()
+	var/obj/item/lighting = locate(/obj/item/lighter) in contents
+	if(lighting)
+		quick_remove_item(lighting, user)
+	else
+		quick_remove_item(/obj/item/clothing/mask/cigarette, user)
+
+/// Removes an item from the packet if there is one
+/obj/item/storage/fancy/cigarettes/proc/quick_remove_item(obj/item/grabbies, mob/user)
+	var/obj/item/finger = locate(grabbies) in contents
+	if(finger)
+		atom_storage.attempt_remove(finger, drop_location())
+		user.put_in_hands(finger)
+
+/obj/item/storage/fancy/cigarettes/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	. = ..()
+	var/obj/item/lighting = locate(/obj/item/lighter) in contents
```

Comment:
`lighter`?

also dont need the variable at all really

Reply:
I honestly don't know why I didn't name the var that
As for why it's there, I don't really know how else to check a list of contents for a type of item

Reply:
I could have sworn you couldn't put `in` in an `if()` with byond

Reply:
it worked, nice. Consider that dealt with then

Reply:
You can its just in has weird precedence if you're doing anything more complicated than this